File: evaluation/benchmark/quicksort2.py
#!/usr/bin/python

import sys

def genlist(max):
  res = []
  for i in range(0,max):
    rev = str(7*i)[::-1]
    res.append(rev)
  return res

def quicksort(arr,left,right):
  store = left
  pivot = arr[right]

  for i in range(left,right):
    if arr[i] < pivot:
      temp = arr[store]
      arr[store] = arr[i]
      arr[i] = temp
      store = store + 1

  arr[right] = arr[store]
  arr[store] = pivot

  if (left < store-1):
    quicksort(arr,left,store-1)
  if (right > store+1):
    quicksort(arr,store+1,right)

n = 1000
if (len(sys.argv) >= 2):
  n = int(sys.argv[1])
strings = genlist(n)
quicksort(strings,0,len(strings)-1)
for s in strings:
  print s

File: evaluation/benchmark/quicksort2.pl
#!/usr/bin/perl -w

sub genlist
{
  my ($max) = @_;
  my @res = ();

  for (my $i = 0; $i < $max; $i++) {
    my $str = 7*$i;
    my $rev = reverse($str);
    push(@res,$rev);
  }

  return @res;
}

sub quicksort
{
  my ($arr,$left,$right) = @_;

  my $store = $left;
  my $pivot = $arr->[$right];
  for (my $i = $left; $i < $right; $i++) {
    if ($arr->[$i] lt $pivot) {
      my $temp = $arr->[$store];
      $arr->[$store] = $arr->[$i];
      $arr->[$i] = $temp;
      $store++;
    }
  }
  $arr->[$right] = $arr->[$store];
  $arr->[$store] = $pivot;

  if ($left < $store-1) {
    quicksort($arr,$left,$store-1);
  }
  if ($right > $store+1) {
    quicksort($arr,$store+1,$right);
  }
}

my $n = ($#ARGV < 0) ? 1000 : $ARGV[0];
my @strings = genlist($n);
my @sorted = @{quicksort(\@strings,0,$#strings)};
foreach $str (@strings) {
  print "$str\n";
}
